### PR TITLE
fix: `.idea/modules.xml` should always uses `/` instead of `\`

### LINKED ISSUE #581: fix:  `.idea/modules.xml` should not produce `\` on Windows
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Version

3.1.1

### Description

When you bootstrap on Windows, `.idea/modules.xml` will be updated to Windows-style path separators. This is unexpected because IDEA will produce `/` on whatever platforms, resulting in file changes with version control.

### Steps to reproduce

1. Create a project that uses *melos* on macOS or Linux.
2. Commit it to version control.
3. Bootstrap the project on Windows.

### Expected behavior

`/` should be always presenting rather than a platform-specific separator.

### Screenshots

image

### Additional context and comments

https://github.com/invertase/melos/blob/b049ed897402921a5b0f3b818e49b47e3b3bf4cf/packages/melos/lib/src/common/intellij_project.dart#L152-L154

Replacing `\` with `/` should fix the issue.

### PR DESCRIPTION
**Pull request by AlexV525**
Resolves #581.

## Description

Replace all `\` with `/` regardless of platforms.

## Type of Change

- [ ] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [x] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

File: packages/melos/test/common/intellij_project_test.dart [hunk 1]
```diff
@@ -29,4 +29,29 @@ void main() {
...
+          '''
+          name: test
+          ''',
+          path: 'test',
+        );
+      final workspace = workspaceBuilder.build();
+      final project = IntellijProject.fromWorkspace(workspace);
+      await project.generate();
+      final modulesXml = readTextFile(project.pathModulesXml);
+      expect(modulesXml, contains(r'file://$PROJECT_DIR$/test/melos_test.iml'));
```
Comment (AlexV525) [inline packages/melos/test/common/intellij_project_test.dart:54]: This will be `file://$PROJECT_DIR$/test\melos_test.iml` on Windows without the fix.

**Comment by AlexV525** [comment]
cc @blaugold

**Review by Salakar — APPROVED** [review]

**Comment by Salakar** [comment]
@AlexV525  re you able to sync with `main` and then I can merge this, thanks

**Comment by AlexV525** [comment]
> @AlexV525 re you able to sync with `main` and then I can merge this, thanks

@Salakar Done. It can be done once you've setup an update button in the repo setting and you'll be not required to ask for updates. :)

**Comment by Salakar** [comment]
> > @AlexV525 re you able to sync with `main` and then I can merge this, thanks
> 
> @Salakar Done. It can be done once you've setup an update button in the repo setting and you'll be not required to ask for updates. :)

I wondered where the option vanished to 😫 thanks
image

**Comment by AlexV525** [comment]
> I wondered where the option vanished to 😫 thanks image

This should be it.

**Pull request merged**